internal/cloud_providers/aws/aws.go

```diff
@@ -0,0 +1,160 @@
+package cloudprovider
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+)
+
+const (
+	// Default Region for AWS CLI
+	DefaultAWSRegion = "eu-west-1"
+)
+
+type AWSConnection struct {
+	credentials  *credentials.Credentials
+	awsConfig    *aws.Config
+	awsSession   *session.Session
+	EC2          *AWSEC2Connection
+	Route53      *AWSRoute53Connection
+	STS          *AWSSTSConnection
+	CostExplorer *AWSCostExplorerConnection
+	accountID    string
+	user         string
+	password     string
+	region       string
+}
+
+// AWSConnectionOption defines the options for creating different sets of AWS services connections
+type AWSConnectionOption func(*AWSConnection)
+
+// NewAWSConnection creates a connection with AWS APIs. Based on the AWSConnectionOptions, it will create different clients for every available service
+func NewAWSConnection(user string, password string, region string, opts ...AWSConnectionOption) (*AWSConnection, error) {
+	var token string = ""
+
+	// If there's no region specified, it will take the default one.
+	if region == "" {
+		region = DefaultAWSRegion
+	}
+
+	// Third argument (token) it's not used. For more info check docs: https://pkg.go.dev/github.com/aws/aws-sdk-go/aws/credentials#NewStaticCredentials
+	credentials := credentials.NewStaticCredentials(user, password, token)
+	if credentials == nil {
+		return nil, fmt.Errorf("Cannot load StaticCredentials for AWS Cloud Provider")
+	}
+
+	conn := &AWSConnection{
```

Comment:
can be simplified with
```go
conn := &AWSConnection{
credentials: credentials,
region:      region,
user:        user,
password:    password,
}
```
for other fields the default value will be applied
https://go.dev/ref/spec#The_zero_value